Add jump start guide

PawelLipski opened the pull request.

PawelLipski commented:
@Andrea @kailuowang please take a look, that's basically the blog post https://virtuslab.com/blog/cats-library/, let me know if you think this needs any modifications

Andrea commented:
Probably better that this PR is yours, I was basically re writing it, no point

LukaJCB commented:
Hi @PawelLipski, does it make sense to update this to 1.0? I don't think having a jump start guide for 0.9 is in the best interest here if I'm being honest and I don't think anything that's particular to 0.9 from the code portion. Anyways thanks much for the great work!

kailuowang commented:
Thanks for the great work. To be honest, I am not sure how I feel about having `Future` in the jump start guide.

PawelLipski commented:
@LukaJCB Sure, I'll migrate to 1.0.0... I'm sure the Cartesian syntax changed in the meantime, I'll check the other pieces as well.
@kailuowang Yes, I'm aware of this problem... still, it's kind of most likely for typical users to employ Cats for dealing with futures and slick DBIOs. And cats provide instances for Future out of the box... so I guess it's perfectly OK to base the jump start guide on Futures.

PawelLipski commented:
@LukaJCB I added a commit that migrates the guide to 1.0.1. In particular, I replaced `|@|` syntax with `mapN`, removed the remarks about `traverseU`/`sequenceU` and renamed `EitherT.liftT` to `EitherT.liftF`. I can't see any other things that need to be updated... pls let me know if you spot anything.

PawelLipski commented:
Also, will soon link the new guide so that's accesible in the website.

PawelLipski commented:
@Andrea @kailuowang @LukaJCB I fixed all the issues the I could see so far, pls review

Andrea commented on build.sbt:
```diff
@@ -102,6 +102,7 @@ lazy val includeGeneratedSrc: Setting[_] = {
 lazy val catsSettings = commonSettings ++ publishSettings ++ scoverageSettings ++ javadocSettings
 
 lazy val scalaCheckVersion = "1.13.5"
+lazy val scalaMockVersion = "3.6.0"
```
Is it just me that finds mocks (in a functional context) a bit of a flag for things that could be improved?

LukaJCB replied:
I fully agree, I think `Future` is good as a gateway drug, but I'm not sure we should encourage mocks. :)

PawelLipski replied:
Okey, fixing now

kailuowang commented:
Thanks very much! This is looking good. We need a link to it, how about an entry close to the top in the FAQs LIst?

kailuowang commented:
I propose we add it as a new FAQ question and place it second on the list (after what import do I need). 
The question could be something like 
"I am new to pure functional programming, what quick wins can I get from Cats?"

PawelLipski commented:
Okey, I added to FAQ as suggested and removed from main menu. It's not linked now from anywhere else but that FAQ answer, though.

PawelLipski commented:
@kailuowang do you see any outstanding issues with this PR and can we think of merging? Recent travis fail is apparently due to some OOM... https://api.travis-ci.org/v3/job/374505453/log.txt

PawelLipski commented:
@LukaJCB can you merge? ;) or is there anything specific that holds us back from merging? ^^

LukaJCB commented:
I keep saying this, but yeah sorry for the delay and thanks for your work! It is appreciated! :)

PawelLipski commented:
No problem and thank you! :) When (circa) is it going to be published on the website? I guess that's gonna be included in some 1.1.1/1.2.0 right?
